pallets/permissions/src/lib.rs

```diff
@@ -147,6 +161,9 @@ where
         _: &DispatchInfoOf<Self::Call>,
         _: usize,
     ) -> Result<Self::Pre, TransactionValidityError> {
+        #[cfg(feature = "testing")]
+        CALL_PERMS_CHECKED.store(false, Relaxed);
```

Comment:
If the Atomic is changed to a counter, then the assert/panic could be moved here.  A counter would also show if the check is done more than two times.

Since this check will only work for tests that go through the signed extensions, then having the assert here will also allow us to convert tests one-by-one.